azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/domain/StorageService.java

```diff
@@ -0,0 +1,208 @@
+         return UNRECOGNIZED;
+      }
+
+   }
+
+   public enum Status {
+
+       Creating,
+       Created,
+       Deleting,
+       Deleted,
+       Changing,
+       ResolvingDns,
+       Succeeded,
+       UNRECOGNIZED;
+
+      public static Status fromString(final String text) {
+         if (text != null) {
+            for (Status status : Status.values()) {
+               if (text.equalsIgnoreCase(status.name())) {
+                  return status;
+               }
+            }
+         }
+         return UNRECOGNIZED;
+      }
+   }
+
+   @AutoValue
+   public abstract static class StorageServiceProperties {
+
+       /**
+        * Specifies whether the account supports locally-redundant storage, geo-redundant storage, zone-redundant
+        * storage, or read access geo-redundant storage.
+        */
+       public abstract AccountType accountType();
+
+       /**
+        * Specifies the time that the storage account was created.
+        */
+       public abstract Date creationTime();
+
+       /**
+        * Specifies the endpoints of the storage account.
+        */
+       @Nullable
+       public abstract Map<String, String> primaryEndpoints();
+
+       /**
+       * A primaryLocation for the storage account.
+       */
+      @Nullable
+      public abstract String primaryLocation();
+
+      /**
+       * provisioningState for the storage group
+       */
+      @Nullable
+      public abstract Status provisioningState();
+
+       /**
+        * Specifies the secondary endpoints of the storage account.
+        */
+       @Nullable
+       public abstract Map<String, String> secondaryEndpoints();
+
+      /**
+       * Secondary location for the storage group
+       */
+      @Nullable
+      public abstract String secondaryLocation();
+
+      /**
+       * The status of primary endpoints
+       */
+      @Nullable
+      public abstract RegionStatus statusOfPrimary();
+
+      /**
+       * The secondary status of the storage account.
+       */
+      @Nullable
+      public abstract RegionStatus statusOfSecondary();
+
+
+      @SerializedNames({"accountType", "creationTime", "primaryEndpoints",  "primaryLocation",
+              "provisioningState", "secondaryEndpoints", "secondaryLocation", "statusOfPrimary", "statusOfSecondary"})
+      public static StorageServiceProperties create(final AccountType accountType, final Date creationTime,
+              final Map<String, String> primaryEndpoints, final String primaryLocation, final Status provisioningState,
+              final Map<String, String> secondaryEndpoints, final String secondaryLocation,
+              final RegionStatus statusOfPrimary, final RegionStatus statusOfSecondary) {
+
+         return new AutoValue_StorageService_StorageServiceProperties(accountType, creationTime,
+                 primaryEndpoints, primaryLocation, provisioningState,
+                 secondaryEndpoints, secondaryLocation, statusOfPrimary, statusOfSecondary);
```

Comment:
Do we want to have null maps here? Can't we follow the same pattern and default to empty ones?